test/mock/grpc/mock_management_plane_grpc_server.go

```diff
@@ -105,5 +140,23 @@ func (mgs *ManagementGrpcServer) UpdateDataPlaneHealth(
 }
 
 func (mgs *ManagementGrpcServer) Subscribe(in v1.CommandService_SubscribeServer) error {
-	return nil
+	for {
+		request := <-mgs.requestChan
+
+		slog.Debug("Subscribe", "request", request)
+
+		err := in.Send(request)
+		if err != nil {
+			slog.Error("Failed to send management request", "error", err)
+		}
+
+		dataPlaneResponse, err := in.Recv()
+		if err != nil {
+			slog.Error("Failed to receive data plane response", "error", err)
+		} else {
+			mgs.dataPlaneResponsesMutex.Lock()
+			mgs.dataPlaneResponses = append(mgs.dataPlaneResponses, dataPlaneResponse)
+			mgs.dataPlaneResponsesMutex.Unlock()
+		}
```

Comment:
Should we have a separate goroutine for listening to responses, or is it overkill seeing as this is a mock server?

Right now we could technically be buildling up requests in `mgs.requestChan` but not sending them to the subscription because we've not yet received a response from the data plane.

Reply:
yeah thats the correct behaviour. The requestChan acts as queue. If we send a config apply request for example, we shouldn't send a second one until we have got a response back from the agent saying its done handling the first request

Reply:
Roger that, then let's keep it the way it is 👌